lib/rules/require-parentheses-around-arrow-param.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Requires parentheses around arrow function expressions with a single parameter.
+ *
+ * Type: `Boolean`
+ *
+ * Value: `true`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireParenthesesAroundArrowParam": true
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * [1, 2, 3].map((x) => x * x);
+ * // params are always required for multiple parameters
+ * [1, 2, 3].map((x, y, z) => x * x);
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * [1, 2, 3].map(x => x * x);
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(options) {
+        assert(
+            options === true,
+            this.getOptionName() + ' option requires a true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requireParenthesesAroundArrowParam';
+    },
+
+    check: function(file, errors) {
+        if (file.getDialect() !== 'es6') {
```

Comment:
I guess not because it won't error anyway... I can remove it from the new rules.

You can assume that if someone is using an es6 rule they would have esnext enabled but it's always nice to have things that help the user experience - don't know if we could have an error if someone enables an es6 rule without esnext.